firmware/build/extract-mrenclave

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env bash
+
+function print_usage() {
+    echo "Usage: $0 <signed_enclave> [config_file]"
+    echo "  or   $0 <unsigned_enclave> <config_file>"
+    echo ""
+    echo "Options:"
+    echo "  signed_enclave:    path of a signed enclave binary file."
+    echo "  unsigned_enclave:  path of an unsigned enclave binary file."
+    echo "  config_file:       configuration file specifying the enclave properties."
+    echo "                     refer to the oesign sign --help for the list of properties."
+    echo "                     this option is only required for unsigned enclaves."
+    echo ""
+    echo "Description:"
+    echo "  This script extracts the MRENCLAVE and the DIGEST values from the enclave"
+    echo "  binary and prints them to stdout. The script can be used both for unsigned"
+    echo "  and signed enclave binaries."
+    echo ""
+    echo "  Signed binaries:"
+    echo "    The MRENCLAVE and DIGEST are calculated from the signed enclave binary and"
```

Comment:
It does, it takes precedence over what is written in the binary. From `oesign` documentation:

```
  The properties are read from the CONFIG_FILE. They override any
  properties that were already defined inside the enclave image
  through use of OE_SET_ENCLAVE_SGX or OE_SET_ENCLAVE_SGX2 macros.
  If a property is not explicitly set, the default values override
  the definitions set by SET_OE_ENCLAVE_SGX(2) macros.
  These properties include:
  ```
  
I verified the source code and it is in fact used in the same way if provided for the `digest` option as well. Otherwise, only the information that is already in the binary is used.

In practice, this makes no difference for us, since the `.conf` file is part of our source tree and is the same file we provide to `oesign` to generate the signed binary, so both the signed binary and the `.conf` file contain the exact same information.

The real reason I left this as optional is because this is how `oesign` handles this parameter, so preserving the behavior keeps the script cleaner (less if/elses). Off course, this also adds flexibility to the script and doesn't hide this functionality in case we find out we will need it in the future.

Reply:
That is none of our current use cases. I would actually add that if the binary has a `.signed` extension, then disallow the configuration file parameter. In that way we prevent misusage of our tooling. If anyone wants the additional functionality, they can always use `oesign` directly.

Reply:
Updated. See the last commit